### PR TITLE
Promotions server should now be polled from payments screen without having payments enabled.

### PR DESCRIPTION
**Pull request by jasonrsadler**
Addon fix for #14616 

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request [as needed](https://github.com/brave/browser-laptop/wiki/Pull-request-process).
- [ ] Request a security/privacy review [as needed](https://github.com/brave/handbook/blob/master/development/security.md#how-to-request-a-security-review). (Ask a Brave employee to help if you cannot access this document.)

## Test Plan:

In addition to Test Plan on #14621:

Whether payments is enabled or not, UGP polling should still be present in console when navigating to about:preferences#payments and LEDGER_VERBOSE=true

## Reviewer Checklist:

- [ ] Request a security/privacy review [as needed](https://github.com/brave/handbook/blob/master/development/security.md#how-to-request-a-security-review) if one was not already requested.

Tests

- [ ] Adequate test coverage exists to prevent regressions
- [ ] Tests should be independent and work correctly when run individually or as a suite [ref](https://github.com/brave/browser-laptop/wiki/Code-Guidelines#test-dependencies)
- [ ] New files have MPL2 license header

### CODE REVIEW

File: app/browser/api/ledger.js [hunk 1]
```diff
@@ -182,16 +182,17 @@ const paymentPresent = (state, tabId, present) => {
...
   }
-
-  if (getSetting(settings.PAYMENTS_ENABLED) && present) {
-    if (!balanceTimeoutId) {
-      module.exports.getBalance(state)
-    }
-
+  if (present) {
     appActions.onPromotionGet()
+    clearTimeout(togglePromotionTimeoutId)
```
Comment (ryanml) [inline app/browser/api/ledger.js]: *nit*

Let's wrap this in an `if` statement (`if (togglePromotionTimeoutId)`)

Comment (jasonrsadler) [reply app/browser/api/ledger.js]: Oh! Not a nit for me. Thanks.

**Comment by jasonrsadler** [comment]
Enable wallet notification is remaining and stacking with promo notification. Reverting to wip.

**Comment by jasonrsadler** [comment]
master: 7cb4060
0.23.x: cbd9a0e9d
0.24.x: 600d7c3fb

**Pull request merged**